src/relay/transforms/merge_composite.cc

```diff
@@ -122,7 +122,9 @@ class MergeCompositeWrapper : public ExprMutator {
       Expr new_arg;
       if (arg->IsInstance<CallNode>()) {
         // if we've already processed this call node, return the previous result
-        if (call_map->find(arg) != call_map->end()) {
+        if (call_map->find(arg) != call_map->end() &&
+            ExtractPattern(Downcast<Call>(arg), Downcast<Call>(root->args[i]), var_map, call_map)
```

Comment:
Would that be clearer if we lift the new `ExtractPattern` checker out of this if-block and combine with the one at L136? It seems to me that those two are the same.

Reply:
Thanks for the suggestion. I've cleaned up `CallNode` handling.